src/fieldDataSource.ts

```diff
@@ -38,23 +38,23 @@ export class DefaultFieldDataSource implements FieldDataSource {
       field.class.includes("ethereum") &&
       field.class.includes("address")
     ) {
-      return await this.ethereumWallet.getAddress();
+      return this.ethereumWallet.getAddress();
     }
 
     if (
       this.bitcoinWallet &&
       field.class.includes("bitcoin") &&
       field.class.includes("address")
     ) {
-      return await this.bitcoinWallet.getNewAddress();
+      return this.bitcoinWallet.getNewAddress();
     }
 
     if (
       this.bitcoinFeeService &&
       field.class.includes("bitcoin") &&
       field.class.includes("feePerByte")
     ) {
-      return await this.bitcoinFeeService.retrieveSatsPerByte();
+      return this.bitcoinFeeService.retrieveSatsPerByte();
```

Comment:
🔧 `retrieveSatsPerByte()` is actually async - OK to remove this?

Reply:
yes, `getData` is `async` so it returns a `Promise`.
We return directly the `Promise` returned by `retrieveSatsPerByte` which is fine.

I am actually removing a TS warning by doing that :)

Reply:
cool